### DISCUSSION TITLE
Any repo for the sandbox/workshops?

### DISCUSSION THREAD

**Discussion by carehart** [vmware-tanzu/community-edition#2204] [General]
Folks, if we have corrections/suggestions for [the sandbox/workshops](https://learn.tanzu.io/), where would we propose/offer PRs for those? I searched here for a typo on the second page of [the first one](https://tanzu.vmware.com/developer/workshops/lab-tce-deploy), "environement", but did not find it here (using github search).

If it may be that those sandboxes do not have their text in github, is there someone or someplace that would want to receive (and would act on) such observations/corrections? Thanks.

**Comment by pnbrown** [comment]
Thanks for asking this here. I'm sorry I missed it, but I believe this issue to now be resolved.

**Comment by carehart** [comment]
OK, but for future reference, if I (or anyone reading this) may find other typos, the main question was where such observations and suggested corrections should be offered. And I was hoping it would be something where we could offer them as a PR (rather than simply raise the observation, to leave someone else to find and correct).

Sorry for my raising this here in "discussions". I probably should have raised it as an "issue" instead, especially since there were no other discussions raised by anyone but your own announcements. :-) Not sure why I did that here. But to be clear, I'm asking if the text of the site might be in any github repo where we can propose corrections as PRs, rather than even just raising 'issues". Thanks.

**Reply by pnbrown** [reply]
The website is under the /docs directory so text fixes/enhancements can be submitted via PR.

**Reply by carehart** [reply]
I appreciate that you're trying to help (I'm doing the same), but no, the workshop site/text is not in the /docs directory (nor in /docs/site). And in fact the typo I had mentioned originally has not been corrected. 

Let me elaborate for you, if you'll indulge me. (Again, I'm just trying to help.)

1] First, as for the typo ("environement"), you can see for yourself that it's still not corrected. Once one is logged into the workshops, and they choose the one I'd mentioned ([Deploying Apps and Microservices with Tanzu Community Edition](https://tanzu.vmware.com/developer/workshops/lab-tce-deploy/)), you can then see the text of the page I refer to [using this link](https://tce-workshops-w01-s2038.tce-vmworld.tanzu-dev.com/workshop/content/exercises/01-deploy-modern-apps). (Of course, in the workshop it shows up in a small frame on the left.) The typo is in the first sentence.

And I had shared first how I couldn't find that page's content, via a search here in the github UI. Given your last comment, I tried to find that page's content in the /docs folder. It's a pretty deep folder (even the "site" folder within it has 1500+ files.) 

2] So I just went ahead now and forked/downloaded the whole repo, and searched the entire codebase locally. Since you said the typo was corrected, I looked for a different word in the page, "veterinarians", but no file in the repo exists with that (or "petclinic", the name of the app in the manifest shown--so that workshop code also is not in the repo, it seems, if we may want to make any suggestions about it.) 

3] Perhaps you may have searched for the TITLE of that first workshop ("Deploying Apps and Microservices with Tanzu Community Edition"), and yes, that text appears in \docs\site\content\resources, within _index.html. But FWIW that is merely in an href link to the (hard-coded) URL, https://tanzu.vmware.com/developer/workshops/lab-tce-deploy. Again, the text of that lab-tce-deploy page does not appear in the repo, so really it's not correct to say that "the website" is there.

So that's really 2 things: the text of pages on the site (like that lab-tce-deploy front page), then the text of the subsequent pages WITHIN the workshops (or 3, if we want to count the manifest code separately). Of course, if it may be that someone has decided the don't want to offer that site and workshop content in github, I'd understand (I'd lament it, but I'd understand).

But since you thought it was--and especially since you thought that the typo had been corrected, I am sharing all this to try to help. Again, sorry it's in a discussion rather than an issue.

**Reply by pnbrown** [reply]
I apologize. I misunderstood. You're correct, the workshop site/text is not in the docs directory. I thought you were referring to typos in the project's site and not the workshop when you mentioned future reference. I did not mean to create that extra work for you.

I can pass this feedback along to the folks that manage the workshop site to get those errors corrected. As far as I know the workshops themselves and the site that hosts the labs is not publicly accessible on GitHub. I'll see if there will be a process for collecting feedback and suggestions for the labs.

Thank you for the detail you shared. It's very helpful.

**Reply by carehart** [reply]
Great, thanks. Looking forward to any news on that. To be clear, I don't mind "doing the work" if it helped to advance the conversation. :-) And indeed though the workshop text may never be offered here, I acknowledged that in my first post hoping there may at least be "_someone or someplace that would want to receive (and would act on) such observations/corrections_".

Any news on that front, at least?

**Reply by pnbrown** [reply]
Tagging in @Boskey

**Reply by pnbrown** [reply]
I believe the repos for the two TCE workshops are [here](https://github.com/Boskey/tce-dev-guide) and [here](https://github.com/Boskey/tce-package-guide). 

I'm still working on guidance for the best ways for collecting feedback and the extent to which workshops can or will be community driven.

**Reply by carehart** [reply]
Thanks. Yep, those links to the repos for the workshops are indeed correct, and in fact I can see that the owner, @Boskey, did correct the typo I mention above in [this commit 7 days ago](https://github.com/Boskey/tce-dev-guide/commit/52a641784727c78a9bbc230def2cd51279aa814d#diff-2ddd7c5bd0dfe1b2127ea27228ba838c4a9705363c245a8a5bc5b152eec5847f). That's good to see--though a comment here could have helped us. :-) 

I'll assume that we can offer feedback and suggestions there, though I notice there are no PRs or issues--open or closed--in the repo. (She has previously made other typo changes and updates via direct commits.) If she may want to comment here to clarify, perhaps other future potential contributors may find this thread, and those links.  Or we can just post PRs there and see how things go. :-)

Again, thanks for the followup on all this.

**Reply by pnbrown** [reply]
Looping in @bryanfriedman as well.

**Reply by bryanfriedman** [reply]
Thanks for looping me in. I'll work on getting some clarity on how to continue to collect feedback.

**Reply by pnbrown** [reply]
Thanks @bryanfriedman.
